Editor/Editors/CinemachineVirtualCameraEditor.cs

```diff
@@ -382,25 +378,57 @@ bool useSimple
             }
         }
 
+        List<CinemachineComponentBase> GetSortedPipelineComponentsForCamera(UnityEngine.Object targetObject)
+        {
+            var camera = targetObject as CinemachineVirtualCamera;
+            if (camera == null)
+                return null;
+
+            var cameraComponents = new List<CinemachineComponentBase>();
```

Comment:
This code is creating a lot of GC allocs, which are very bad for editor performance.

Reply:
You can replace the whole thing with this, which is much more efficient:

```
        void GetPipelineTypes(CinemachineVirtualCamera vcam, ref Type[] types)
        {
            for (int i = 0; i < types.Length; ++i)
                types[i] = null;
            if (vcam != null)
            {
                var components = vcam.GetComponentPipeline();
                for (int j = 0; j < components.Length; ++j)
                    types[(int)components[j].Stage] = components[j].GetType();
            }
        }

        // scratch buffers for pipeline types
        Type[] s_PipelineTypeCache0 = new Type[Enum.GetValues(typeof(CinemachineCore.Stage)).Length];
        Type[] s_PipelineTypeCacheN = new Type[Enum.GetValues(typeof(CinemachineCore.Stage)).Length];

        void UpdateStageDataTypeMatchesForMultiSelection()
        {
            for (int i = 0; i < m_hasSameStageDataTypes.Length; ++i)
                m_hasSameStageDataTypes[i] = true;

            if (targets.Length > 1)
            {
                GetPipelineTypes(serializedObject.targetObjects[0] as CinemachineVirtualCamera, ref s_PipelineTypeCache0);
                for (int i = 1; i < targets.Length; ++i)
                {
                    GetPipelineTypes(serializedObject.targetObjects[i] as CinemachineVirtualCamera, ref s_PipelineTypeCacheN);
                    for (int j = 0; j < s_PipelineTypeCache0.Length; ++j)
                        if (s_PipelineTypeCache0[j] != s_PipelineTypeCacheN[j])
                            m_hasSameStageDataTypes[j] = false;
                }
            }
        }
```